product-service/src/main/java/com/extremecoder/productservice/utils/ResponseBuilder.java

```diff
@@ -0,0 +1,80 @@
+package com.extremecoder.productservice.utils;
+
+import com.extremecoder.productservice.dto.ErrorResponseDto;
+import com.extremecoder.productservice.dto.Response;
+import org.springframework.http.HttpStatus;
+import org.springframework.validation.BindingResult;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public final class ResponseBuilder {
+    private ResponseBuilder() {
+    }
+
+    private static List<ErrorResponseDto> getCustomError(BindingResult result) {
+        List<ErrorResponseDto> dtoList = new ArrayList<>();
+        result.getFieldErrors().forEach(fieldError -> {
+            ErrorResponseDto errorResponseDto = ErrorResponseDto.builder()
+                    .field(fieldError.getField())
+                    .message(fieldError.getDefaultMessage())
+                    .build();
+            dtoList.add(errorResponseDto);
+        });
+        return dtoList;
+    }
+
+    public static Response getFailureResponse(BindingResult result, String message) {
+        return Response.builder()
+                .message(message)
+                .errors(getCustomError(result))
+                .status(HttpStatus.BAD_REQUEST.getReasonPhrase())
+                .statusCode(HttpStatus.BAD_REQUEST.value())
+                .timeStamp(LocalDateTime.now())
+                .build();
+    }
+
+    public static Response getFailureResponse(HttpStatus status, String message) {
+        return Response.builder()
+                .message(message)
+                .status(status.getReasonPhrase())
+                .status(HttpStatus.BAD_REQUEST.getReasonPhrase())
+                .statusCode(HttpStatus.BAD_REQUEST.value())
+                .timeStamp(LocalDateTime.now())
+                .build();
+    }
+
+    public static Response getSuccessResponse(HttpStatus status, String message, Object content) {
+        return Response.builder()
+                .message(message)
+                .status(status.getReasonPhrase())
+                .statusCode(status.value())
+                .content(content)
+                .timeStamp(LocalDateTime.now())
+                .build();
+    }
+
+    public static Response getSuccessResponse(HttpStatus status, String message, Object content, int numberOfElement) {
+        return Response.builder()
```

Comment:
Instead of writing line number 59 to 66 we could directly write: 
```
return getSuccessResponse(status, message, content).numberOfElement(numberOfElement).build()
```